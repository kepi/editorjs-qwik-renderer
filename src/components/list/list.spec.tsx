import { createDOM } from "@builder.io/qwik/testing";
import { describe,expect,it } from "vitest";
import List, { type ListBlockData } from './list';

describe('List', async () => {
  describe.each([
    {style: 'unordered', tag: '<ul>'},
    {style: 'ordered', tag: '<ol>'},]
  )('when receives a flat $style list block', ({style, tag}) => {
    const data: ListBlockData = {
      // @ts-expect-error
      style,
      items: [
        'It is a block-styled editor',
        'It returns clean <b>data output</b> in JSON',
        'Designed to be extendable and pluggable with a simple API',
      ],
    };


    it(`renders a ${tag} block`, async () => {
      const { screen, render } = await createDOM();
      await render(<List data={data} />)
      expect(screen.innerHTML).toMatchSnapshot()
    });
  });

  describe.each([
    {style: 'unordered', tag: '<ul>'},
    {style: 'ordered', tag: '<ol>'},]
  )('when receives a nested $style list block', ({style, tag}) => {
    const data: ListBlockData = {
      // @ts-expect-error
      style,
      items: [
        {
          content: 'It is a block-styled editor',
          items: [],
        },
        {
          content: 'It returns clean data output in JSON',
          items: [
            {
              content: 'Designed to be extendable and pluggable with a simple API',
              items: [
                {
                  content: 'Red',
                  items: [],
                },
                {
                  content: 'Green',
                  items: [],
                },
              ],
            },
          ],
        },
        {
          content: 'It returns clean data output in JSON',
          items: [],
        },
        {
          content: 'Designed to be extendable and pluggable with a simple API',
          items: [],
        },
      ],
    };

    it(`renders a ${tag} block`, async () => {
      const { screen, render } = await createDOM();
      await render(<List data={data} />);
      expect(screen.innerHTML).toMatchSnapshot()
    });

    describe('and when className is provided', async () => {
      it.each(['list', 'list-ul px-3 py-2'])(`renders class to all ${tag} blocks`, async (className) => {
        const { screen, render } = await createDOM();
        await render(<List data={data} class={className} />)
        expect(screen.innerHTML).toMatchSnapshot()
      });
    });
  });
});
