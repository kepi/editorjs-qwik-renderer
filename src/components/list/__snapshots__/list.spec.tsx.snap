// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`List > when receives a flat 'ordered' list block > renders a <ol> block 1`] = `"<!--qv --><!--qv --><ol><!--qv --><li><!--qv q:key q:sref=2 q:s--><span>It is a block-styled editor</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=3 q:s--><span>It returns clean <b>data output</b> in JSON</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=4 q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--/qv--></li><!--/qv--></ol><!--/qv--><!--/qv-->"`;

exports[`List > when receives a flat 'unordered' list block > renders a <ul> block 1`] = `"<!--qv --><!--qv --><ul><!--qv --><li><!--qv q:key q:sref=2 q:s--><span>It is a block-styled editor</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=3 q:s--><span>It returns clean <b>data output</b> in JSON</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=4 q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--/qv--></li><!--/qv--></ul><!--/qv--><!--/qv-->"`;

exports[`List > when receives a nested 'ordered' list block > and when className is provided > renders class to all <ol> blocks 1`] = `"<!--qv --><!--qv --><ol class=\\"list\\"><!--qv --><li><!--qv q:key q:sref=2 q:s--><span>It is a block-styled editor</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=3 q:s--><span>It returns clean data output in JSON</span><!--qv --><ol class=\\"list\\"><!--qv --><li><!--qv q:key q:sref=5 q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--qv --><ol class=\\"list\\"><!--qv --><li><!--qv q:key q:sref=7 q:s--><span>Red</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=8 q:s--><span>Green</span><!--/qv--></li><!--/qv--></ol><!--/qv--><!--/qv--></li><!--/qv--></ol><!--/qv--><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=9 q:s--><span>It returns clean data output in JSON</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=a q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--/qv--></li><!--/qv--></ol><!--/qv--><!--/qv-->"`;

exports[`List > when receives a nested 'ordered' list block > and when className is provided > renders class to all <ol> blocks 2`] = `"<!--qv --><!--qv --><ol class=\\"list-ul px-3 py-2\\"><!--qv --><li><!--qv q:key q:sref=2 q:s--><span>It is a block-styled editor</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=3 q:s--><span>It returns clean data output in JSON</span><!--qv --><ol class=\\"list-ul px-3 py-2\\"><!--qv --><li><!--qv q:key q:sref=5 q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--qv --><ol class=\\"list-ul px-3 py-2\\"><!--qv --><li><!--qv q:key q:sref=7 q:s--><span>Red</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=8 q:s--><span>Green</span><!--/qv--></li><!--/qv--></ol><!--/qv--><!--/qv--></li><!--/qv--></ol><!--/qv--><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=9 q:s--><span>It returns clean data output in JSON</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=a q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--/qv--></li><!--/qv--></ol><!--/qv--><!--/qv-->"`;

exports[`List > when receives a nested 'ordered' list block > renders a <ol> block 1`] = `"<!--qv --><!--qv --><ol><!--qv --><li><!--qv q:key q:sref=2 q:s--><span>It is a block-styled editor</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=3 q:s--><span>It returns clean data output in JSON</span><!--qv --><ol><!--qv --><li><!--qv q:key q:sref=5 q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--qv --><ol><!--qv --><li><!--qv q:key q:sref=7 q:s--><span>Red</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=8 q:s--><span>Green</span><!--/qv--></li><!--/qv--></ol><!--/qv--><!--/qv--></li><!--/qv--></ol><!--/qv--><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=9 q:s--><span>It returns clean data output in JSON</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=a q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--/qv--></li><!--/qv--></ol><!--/qv--><!--/qv-->"`;

exports[`List > when receives a nested 'unordered' list block > and when className is provided > renders class to all <ul> blocks 1`] = `"<!--qv --><!--qv --><ul class=\\"list\\"><!--qv --><li><!--qv q:key q:sref=2 q:s--><span>It is a block-styled editor</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=3 q:s--><span>It returns clean data output in JSON</span><!--qv --><ul class=\\"list\\"><!--qv --><li><!--qv q:key q:sref=5 q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--qv --><ul class=\\"list\\"><!--qv --><li><!--qv q:key q:sref=7 q:s--><span>Red</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=8 q:s--><span>Green</span><!--/qv--></li><!--/qv--></ul><!--/qv--><!--/qv--></li><!--/qv--></ul><!--/qv--><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=9 q:s--><span>It returns clean data output in JSON</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=a q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--/qv--></li><!--/qv--></ul><!--/qv--><!--/qv-->"`;

exports[`List > when receives a nested 'unordered' list block > and when className is provided > renders class to all <ul> blocks 2`] = `"<!--qv --><!--qv --><ul class=\\"list-ul px-3 py-2\\"><!--qv --><li><!--qv q:key q:sref=2 q:s--><span>It is a block-styled editor</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=3 q:s--><span>It returns clean data output in JSON</span><!--qv --><ul class=\\"list-ul px-3 py-2\\"><!--qv --><li><!--qv q:key q:sref=5 q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--qv --><ul class=\\"list-ul px-3 py-2\\"><!--qv --><li><!--qv q:key q:sref=7 q:s--><span>Red</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=8 q:s--><span>Green</span><!--/qv--></li><!--/qv--></ul><!--/qv--><!--/qv--></li><!--/qv--></ul><!--/qv--><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=9 q:s--><span>It returns clean data output in JSON</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=a q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--/qv--></li><!--/qv--></ul><!--/qv--><!--/qv-->"`;

exports[`List > when receives a nested 'unordered' list block > renders a <ul> block 1`] = `"<!--qv --><!--qv --><ul><!--qv --><li><!--qv q:key q:sref=2 q:s--><span>It is a block-styled editor</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=3 q:s--><span>It returns clean data output in JSON</span><!--qv --><ul><!--qv --><li><!--qv q:key q:sref=5 q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--qv --><ul><!--qv --><li><!--qv q:key q:sref=7 q:s--><span>Red</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=8 q:s--><span>Green</span><!--/qv--></li><!--/qv--></ul><!--/qv--><!--/qv--></li><!--/qv--></ul><!--/qv--><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=9 q:s--><span>It returns clean data output in JSON</span><!--/qv--></li><!--/qv--><!--qv --><li><!--qv q:key q:sref=a q:s--><span>Designed to be extendable and pluggable with a simple API</span><!--/qv--></li><!--/qv--></ul><!--/qv--><!--/qv-->"`;
